waku/v2/protocol/waku_archive/driver/postgres_driver.nim

```diff
@@ -0,0 +1,248 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  std/db_postgres,
```

Comment:
I noticed that here we use the standard library PostgreSQL driver [std/db_postgres](https://nim-lang.org/docs/db_postgres.html). This is an idiomatic wrapper library for the PostgreSQL C library `libpq`.

Regarding the request concurrency issue I described above, I think using this library "as is" (without notifying the nim-chronos scheduler) may present the same limitation as the SQLite driver: it will block the scheduler thread not allowing new Waku Store requests to be processed.

Maybe @arnetheduck or @dryajov have an idea of how we could achieve this integration with the nim-chronos async runtime.

Reply:
> achieve this integration with the nim-chronos async runtime.

porting https://github.com/treeform/pg to chronos is probably the way to go to achieve this - it actually looks pretty trivial to do, chances are it would work with minor modifications (to handle cancellation above all)